lib/models/github_repository.dart

```diff
@@ -0,0 +1,12 @@
+import 'package:top_git_rep/models/search_result.dart';
+
+class GithubRepository {
+  const GithubRepository(this.client);
+
+  final GithubRepository client;
+
+  Future<SearchResult> search(String term) async {
```

Comment:
Maybe there is a better way to name this method and parameter? Like what is it searching for? and what is term?